### PR TITLE
Improve docs, in particular shortcuts. 

### PR DESCRIPTION
**Pull request by Carreau**
Mostly cleanup and reorder documentation.

### CODE REVIEW

File: IPython/utils/warn.py [hunk 1]
```diff
@@ -10,11 +10,15 @@
 
 import sys
 import warnings
+from IPython.utils.decorators import undoc
 
 warnings.warn("The module IPython.utils.warn is deprecated since IPython 4.0, use the standard warnings module instead", DeprecationWarning)
 
+@undoc
```
Comment (takluyver) [inline IPython/utils/warn.py]: If the whole of this module should be undocumented, it's easier to add it as a skip pattern in `autogen_api.py` than to mark every function as undoc.

File: docs/autogen_shortcuts.py [hunk 2]
```diff
@@ -5,6 +5,10 @@
```
Comment (takluyver) [inline docs/autogen_shortcuts.py]: I haven't had a chance to look into this script yet, so I won't try to review changes to it - at some point I'll look through it as a whole.

File: IPython/utils/warn.py (hunk 1 above)
Comment (Carreau) [inline IPython/utils/warn.py]: > If the whole of this module should be undocumented, it's easier to add it as a skip pattern in autogen_api.py than to mark every function as undoc.

Done.

**Comment by Carreau** [comment]
Thanks !

**Pull request merged**